frontend/src/main/web/lib/components/glossary/DataTable.jsx

```diff
@@ -0,0 +1,438 @@
+
+  _isTranslationSelected: function () {
+    return !StringUtils.isEmptyOrNull(this.props.selectedTransLocale);
+  },
+
+  _getSourceColumn: function() {
+    var self = this, srcLocaleName = "";
+    if(!_.isUndefined(this.props.srcLocale) && !_.isNull(this.props.srcLocale)) {
+      srcLocaleName = this.props.srcLocale.locale.displayName;
+    }
+    return (<Column
+      label={srcLocaleName}
+      key={self.ENTRY.SRC.field}
+      width={150}
+      dataKey={0}
+      flexGrow={1}
+      cellRenderer={self._renderSourceCell}
+      headerRenderer={self._renderSourceHeader}
+      />);
+  },
+
+  _getTransColumn: function() {
+    var self = this;
+    return (<Column
+      label="Translations"
+      key={self.ENTRY.TRANS.field}
+      width={150}
+      dataKey={0}
+      flexGrow={1}
+      cellRenderer={self._renderTransCell}
+      headerRenderer={self._renderTransHeader}
+      />);
+  },
+
+  _getPosColumn: function() {
+    var self = this;
+    return (<Column
+      label="Part of Speech"
+      key={self.ENTRY.POS.field}
+      width={150}
+      dataKey={0}
+      cellRenderer={self._renderPosCell}
+      headerRenderer={self._renderPosHeader}
+      />);
+  },
+
+  _getDescColumn: function() {
+    var self = this;
+    return (<Column
+      label="Description"
+      key={self.ENTRY.DESC.field}
+      width={150}
+      flexGrow={1}
+      dataKey={0}
+      cellRenderer={self._renderDescCell}
+      headerRenderer={self._renderDescHeader}
+      />);
+  },
+
+  _getTransCountColumn: function() {
+    var self = this;
+    return (<Column
+      label="Translations"
+      key={self.ENTRY.TRANS_COUNT.field}
+      width={120}
+      cellClassName="tac"
+      dataKey={0}
+      cellRenderer={self._renderTransCountCell}
+      headerRenderer={self._renderTransCountHeader}
+      />);
+  },
+
+  _getActionColumn: function() {
+    var self = this;
+    return (<Column
+      label=""
+      key="Actions"
+      cellClassName="ph1/4"
+      width={300}
+      dataKey={0}
+      isResizable={false}
+      cellRenderer={self._renderActionCell}
+      />)
+  },
+
+  _onRowMouseEnter: function (event, rowIndex) {
+    if (this.state.hoveredRow !== rowIndex) {
+      this.setState({hoveredRow: rowIndex});
+    }
+  },
+
+  _onRowMouseLeave: function (event, rowIndex) {
+    var unhoveredRow = -1;
+    if (this.state.hoveredRow !== unhoveredRow) {
+      this.setState({hoveredRow: unhoveredRow});
+    }
+  },
+
+  _onRowClick: function (event, rowIndex) {
+    var resId = this._rowGetter(rowIndex)[0];
+    if(this.props.focusedRow.rowIndex !== rowIndex) {
+      Actions.updateFocusedRow(resId, rowIndex);
+    }
+  },
+
+  _onRowBlur: function (event, rowIndex) {
+
+  },
+
+  _rowClassNameGetter: function (rowIndex) {
+    if(this.props.focusedRow && this.props.focusedRow.rowIndex === rowIndex) {
+      return 'bgcsec30a cdtrigger';
+    } else if(this.state.hoveredRow === rowIndex) {
+      return 'bgcsec20a cdtrigger';
+    }
```

Comment:
This returns `undefined` when the row is not focused or hovered. Is that intended?